### PR TITLE
Kd od work

### PR DESCRIPTION
**Pull request by KiranDarbha**

### CODE REVIEW

File: internal/mqini/mqini.go [hunk 1]
```diff
@@ -76,3 +88,168 @@ func GetErrorLogDirectory(qm *QueueManager) string {
 	}
 	return filepath.Join(qm.Prefix, "qmgrs", qm.Directory, "errors")
 }
+
+//Update qm.ini file with the user supplied stanzas.
+func AddStanzas(qmname string) error {
+
+	qm, err := GetQueueManager(qmname)
```
Comment (arthurbarr) [inline internal/mqini/mqini.go]: Formatting looks off in a number of places.  Code needs to be passed through "go fmt" to format it the right way.  Most editors do this, but perhaps not the one you're using.

File: internal/mqini/mqini.go [hunk 2]
```diff
@@ -76,3 +88,168 @@ func GetErrorLogDirectory(qm *QueueManager) string {
...
+		}
+	}
+
+	//append if stanza doesn't exist.
+	if len(sbAppend.String()) > 0 {
+       qmconfigStr = qmconfigStr + sbAppend.String()
+	}
+
+	//If this is a unit-test call, we don't write, just return.
+	if isUnitTest {
```
Comment (arthurbarr) [inline internal/mqini/mqini.go]: Having the production code have a flag for `isUnitTest` doesn't seem right to me.  Couldn't we just refactor the functions a bit, to separate the writing into a separate function?  Then we could unit test all the INI file manipulation as pure Strings, and the prod code could call a wrapper function which does the string manipulation, then writes it to a file.

File: internal/mqini/mqini_test.go [hunk 3]
```diff
@@ -59,6 +62,60 @@ func TestGetQueueManager(t *testing.T) {
 			if d != table.errorLogDir {
 				t.Errorf("Expected error log directory=%v; got %v", table.errorLogDir, d)
 			}
+
```
Comment (arthurbarr) [inline internal/mqini/mqini_test.go]: It seems to me that this is really a few different tests conflated into one.  I'm finding it hard to see if you actually check that the resulting INI file has the right data in it.  By breaking up the code into smaller functions, I wonder if we can have separate Test functions which target the individual units more clearly.

File: internal/mqini/mqini.go [hunk 4]
```diff
@@ -76,3 +88,168 @@ func GetErrorLogDirectory(qm *QueueManager) string {
...
+    filepath.Walk("/etc", func(path string, f os.FileInfo, err error) error {
+        if strings.HasSuffix(path, ".ini") {
+            fileList = append(fileList, path)
+		}
+		return nil       
+	})
+	return fileList
+}
+
+// Based on the ini file(qm.ini or mqs.ini or mqat.ini), return corresponding
```
Comment (arthurbarr) [inline internal/mqini/mqini.go]: Godoc comments usually follow a pattern which includes the name of the function or package your describing.  See [this blog entry on documenting Go code](https://blog.golang.org/godoc-documenting-go-code)

**Comment by arthurbarr** [comment]
Also, the copyright dates on some of the files need to be updated.

File: internal/mqini/mqini.go (hunk 1 above)
Comment (KiranDarbha) [reply internal/mqini/mqini.go]: I have ran the latest commit with gofmt and formatted the code.

File: internal/mqini/mqini.go (hunk 2 above)
Comment (KiranDarbha) [reply internal/mqini/mqini.go]: Thanks, I have now removed the flag in my latest commit.

File: internal/mqini/mqini_test.go (hunk 3 above)
Comment (KiranDarbha) [reply internal/mqini/mqini_test.go]: Thanks, I have rewritten tests to separate them and run as stanza verification tests and also increased coverage.

File: internal/mqini/mqini_test.go [hunk 5]
```diff
@@ -62,3 +64,160 @@ func TestGetQueueManager(t *testing.T) {
...
+		t.Errorf("Unexpected error: [%s]\n", err.Error())
+	}
+	if len(qmConfig) == 0 {
+		t.Errorf("Unexpected stanza file update: qm.ini[%s]\n", atConfig)
+	}
+	if len(atConfig) == 0 {
+		t.Errorf("Expected stanza file not found: mqat.ini\n")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(userconfig))
```
Comment (arthurbarr) [inline internal/mqini/mqini_test.go]: Here, it looks like you're checking that all the lines in the input file are present in one of the output strings, but it's not checking that it's in the _right_ output string.

File: internal/mqini/mqini_test.go [hunk 6]
```diff
@@ -16,7 +16,9 @@ limitations under the License.
 package mqini
```
Comment (arthurbarr) [inline internal/mqini/mqini_test.go]: Copyright of this file hasn't been changed.  Need to add ", 2019".

File: internal/mqini/mqini_test.go (hunk 5 above)
Comment (arthurbarr) [reply internal/mqini/mqini_test.go]: Do you think that would be useful, or do you think this is covered by your other test which ensures the stanzas are directed to the right file?

Comment (KiranDarbha) [reply internal/mqini/mqini_test.go]: Thank you. Yes, I have modified the test to check right strings in right files and I have also removed redundant test.

File: internal/mqini/mqini_test.go (hunk 6 above)
Comment (KiranDarbha) [reply internal/mqini/mqini_test.go]: Thanks. updated it now.

Comment (arthurbarr) [reply internal/mqini/mqini_test.go]: Copyright statements need to be formatted with two dates: the first is the year the code was first available, and the second is the year of the most recent change.  If both years are the same, then you can just have one date.  So this statement needs to be as follows:

```
© Copyright IBM Corporation 2018, 2019
```

Comment (KiranDarbha) [reply internal/mqini/mqini_test.go]: Thanks, I have updated the copy right now.

**Pull request merged**